README.md

```diff
@@ -1,3 +1,111 @@
-# Node.js Express Web Service using standard `node` image
+# Primer: Node.js Application Hardening
 
-TO-DO
+## The Problem and the Solution 
+
+Creating high-quality container images is hard. How to choose an optimal base image? How to configure a multistage build to make the end slim? How to keep the vulnerability scanners calm? One must be a true container expert to get it right. However, there might be an alternative way.
+
+**Automated container image hardening**
+
+* Build an image FROM your favorite base.
+* Instrument the image with our sensors.
+* Run an instrumented container "probing" its functionality.
+* Build a hardened version of the image using collected intelligence.
+
+![image](https://user-images.githubusercontent.com/45476902/218093055-50a44810-db1a-43fd-a71d-909e521d4a55.png)
+
+
+### Prerequisites
+* A fresh version of the Slim CLI [installed and configured](https://portal.slim.dev/cli)
+
+* [A container registry connector configured](https://portal.slim.dev/connectors)
+
+* Docker (Engine or Desktop) is installed locally.
+
+### Demo
+
+Before we move to the steps, let's visualise with this great image from [Ivan](https://github.com/iximiuz) on how the flow looks like!
+![image](https://user-images.githubusercontent.com/45476902/218159028-d2b21334-bfeb-45dd-8d2d-725fbe3d3520.png)
+
+
+#### Step 1: Instrument  🕵️
+
+The first step involves instrumenting the image and simply speaking it means that the Slim engine's sensor is going to act like an intelligence agent (the Container's James Bond) to collect data for the further probe  🕵️
```

Comment:
You know James Bond is known for destroying everything he interacts with, right? :))